From 7492337c1344eb0d51a4f60f687d1aebc3670468 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Fri, 25 Jul 2008 23:06:21 +0100
Subject: [PATCH] fix-pcf50633-remove-charger-curlim-and-enable-apis-from-export.patch

Setting the current limit directly and enabling the charger
isn't anyone's business except pcf50633 driver itself, so these
two functions should not be exported and become static.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/i2c/chips/pcf50633.c |   10 ++++++----
 include/linux/pcf50633.h     |    6 ------
 2 files changed, 6 insertions(+), 10 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index 4c73a9d..a5efb4e 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -194,6 +194,10 @@ EXPORT_SYMBOL_GPL(pcf50633_global);
 
 static struct platform_device *pcf50633_pdev;
 
+static void pcf50633_usb_curlim_set(struct pcf50633_data *pcf, int ma);
+static void pcf50633_charge_enable(struct pcf50633_data *pcf, int on);
+
+
 /***********************************************************************
  * Low-Level routines
  ***********************************************************************/
@@ -1384,7 +1388,7 @@ static DEVICE_ATTR(voltage_hcldo, S_IRUGO | S_IWUSR, show_vreg, set_vreg);
  ***********************************************************************/
 
 /* Set maximum USB current limit */
-void pcf50633_usb_curlim_set(struct pcf50633_data *pcf, int ma)
+static void pcf50633_usb_curlim_set(struct pcf50633_data *pcf, int ma)
 {
 	u_int8_t bits;
 
@@ -1437,7 +1441,6 @@ set_it:
 				PCF50633_MBCC1_AUTORES);
 
 }
-EXPORT_SYMBOL_GPL(pcf50633_usb_curlim_set);
 
 static ssize_t show_usblim(struct device *dev, struct device_attribute *attr,
 			   char *buf)
@@ -1462,7 +1465,7 @@ static ssize_t show_usblim(struct device *dev, struct device_attribute *attr,
 static DEVICE_ATTR(usb_curlim, S_IRUGO | S_IWUSR, show_usblim, NULL);
 
 /* Enable/disable charging */
-void pcf50633_charge_enable(struct pcf50633_data *pcf, int on)
+static void pcf50633_charge_enable(struct pcf50633_data *pcf, int on)
 {
 	u_int8_t bits;
 	u_int8_t usblim;
@@ -1499,7 +1502,6 @@ void pcf50633_charge_enable(struct pcf50633_data *pcf, int on)
 	reg_set_bit_mask(pcf, PCF50633_REG_MBCC1, PCF50633_MBCC1_CHGENA,
 			 bits);
 }
-EXPORT_SYMBOL_GPL(pcf50633_charge_enable);
 
 #if 0
 #define ONE			1000000
diff --git a/include/linux/pcf50633.h b/include/linux/pcf50633.h
index b4d65dc..c11a637 100644
--- a/include/linux/pcf50633.h
+++ b/include/linux/pcf50633.h
@@ -111,12 +111,6 @@ pcf50633_onoff_set(struct pcf50633_data *pcf,
 		   enum pcf50633_regulator_id reg, int on);
 
 extern void
-pcf50633_usb_curlim_set(struct pcf50633_data *pcf, int ma);
-
-extern void
-pcf50633_charge_enable(struct pcf50633_data *pcf, int on);
-
-extern void
 pcf50633_backlight_resume(struct pcf50633_data *pcf);
 
 extern u_int16_t
-- 
1.5.6.3

